apiserver/vshn/postgres/get.go

```diff
@@ -0,0 +1,48 @@
+package postgres
+
+import (
+	v1 "appcat-apiserver/apis/vshn/postgres/v1"
+	"appcat-apiserver/apiserver"
+	"context"
+	"fmt"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+var _ rest.Getter = &vshnPostgresBackupStorage{}
+
+// Get returns a VSHNPostgresBackupStorage service based on stackgres SGBackup resource
+func (v *vshnPostgresBackupStorage) Get(ctx context.Context, name string, _ *metav1.GetOptions) (runtime.Object, error) {
+	namespace, ok := request.NamespaceFrom(ctx)
+	if !ok {
+		return nil, fmt.Errorf("cannot get namespace from resource")
+	}
+
+	instances, err := v.vshnpostgresql.GetVSHNPostgreSQLMetadata(ctx, namespace)
+	if err != nil {
+		return nil, fmt.Errorf("cannot list VSHNPostgreSQL instances")
+	}
+
+	var vshnBackup *v1.VSHNPostgresBackup
+	for k, value := range *instances {
+		backupInfo, err := v.sgbackups.GetSGBackup(ctx, name, string(value))
+		if err != nil {
+			resolvedErr := apiserver.ResolveError(sgbackupGroupVersionResource.GroupResource(), err)
+			if apierrors.IsNotFound(resolvedErr) {
+				continue
+			}
+			return nil, err
+		}
+
+		vshnBackup = v1.NewVSHNPostgresBackup(backupInfo, string(k), namespace)
+	}
```

Comment:
The backup name is composted like that `<db-instance-name>-<timestamp-to-seconds>`. The database instance has already a dynamic part at the end of the chosen name so there's no chance of having two backups with the same name. Ex: `pgsql-app1-prod-sxbmc-2023-02-24-15-05-08`.

Reply:
Hmm I guess that's fine. I'm still not a huge fan as it could still be theoretically possible that there are two SGBackups with the same name and that would lead to a very weird bug, but I'm ok with it

Reply:
Well, theoretically speaking even if we add the namespace to the sgbackup name we will not have eliminated 100% the chance of 2 backups having the same name no? It will just reduce further the possibility of having 2 with the same name :)

Reply:
Well namespace-name is unique, so no. But yeah you'd have to consider some other things